File: backend/app/models/summary.py
from sqlalchemy import Column, Integer, String, ForeignKey, Text
from sqlalchemy.orm import relationship
from app.db.base_class import Base

class Summary(Base):
    id = Column(Integer, primary_key=True, index=True)
    mode = Column(String) # quick, structured, exam, meeting, eli5, executive
    content = Column(Text, nullable=False)
    
    session_id = Column(Integer, ForeignKey("session.id"))
    session = relationship("Session", back_populates="summaries")

File: backend/models/Summary.js
module.exports = (sequelize, DataTypes) => {
  return sequelize.define("Summary", {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    mode: DataTypes.STRING,
    content: DataTypes.TEXT,
    session_id: DataTypes.INTEGER
  }, {
    tableName: "summaries",
    timestamps: false
  });
};
